File: python/bob/bob.py
import re


def response(hey_bob):
    hey_bob = re.sub(r"\s+", '', hey_bob)

    if len(hey_bob) == 0:
        return 'Fine. Be that way!'

    if re.search(r'[A-Z]', hey_bob) and not re.search(r'[a-z]', hey_bob):
        return "Calm down, I know what I'm doing!" if re.search(r'\?$', hey_bob) else 'Whoa, chill out!'

    return 'Sure.' if re.search(r'\?$', hey_bob) else 'Whatever.'

File: erlang/bob/src/bob.erl
-module(bob).

-export([response/1]).

response(S) ->
    TRIMMED = string:trim(S, trailing),
    case TRIMMED of
      [] -> "Fine. Be that way!";
      _ -> response(is_question(TRIMMED), is_yelling(TRIMMED))
    end.

response(true, true) ->
    "Calm down, I know what I'm doing!";
response(true, _) -> "Sure.";
response(_, true) -> "Whoa, chill out!";
response(_, _) -> "Whatever.".

is_question(S) -> $? =:= lists:last(S).
is_yelling(S) ->
    string:uppercase(S) =:= S andalso
      not (string:lowercase(S) =:= S).
